templates/azure/mender-artifact-upload.yml

```diff
@@ -0,0 +1,59 @@
+# Use this template to upload a Mender Artifact to a Mender server.
+#
+# The template is Azure DevOps step template with parameters. It's designed to be included
+# from the GitHub repository.
+#
+# The template has the following parameters:
+# - `mender_uri`:      Mender server URL.
+# - `mender_pat`:      Mender server access token.
+# - `mender_artifact`: Relative to $(System.DefaultWorkingDirectory) path of Mender Artifact file
+
+parameters:
+- name: 'mender_uri'
+  type: string
+  default: 'https://hosted.mender.io'
+- name: 'mender_pat'
+  type: string
+- name: 'mender_artifact'
+  type: string
+
+steps:
+- script: |
+    # input validation
+    if [ -z "${{ parameters.mender_uri }}" ]; then
+      echo "ERROR: mender_uri is not found."
+      exit 1
+    fi
+    if [ -z "${{ parameters.mender_pat }}" ]; then
+      echo "ERROR: mender_pat is not found."
+      exit 1
+    fi
+    if ! ls ${{ parameters.mender_artifact }} &>/dev/null; then
+      echo "ERROR: file ${{ parameters.mender_artifact }} is not found."
+      exit 1
+    fi
+    # check requirements
+    if [ ! -f $(which curl) ]; then
+      echo "ERROR: 'curl' is missing"
+      exit 1
+    fi
+    # Makes management 'upload artifact' API call to Mender server, using curl
```

Comment:
Why not using `mender-cli`? If this is running with our image we should use `mender-cli` where possible.